returnn/datasets/postprocessing.py

```diff
@@ -221,7 +231,20 @@ def _collect_single_seq(self, seq_idx: int) -> Optional[DatasetSeq]:
         while True:
             try:
                 loaded_seq_idx, tensor_dict = next(self._data_iter)
+                self._data_iter_produced_num_seqs += 1
+                if self._num_seqs is not None:
+                    assert self._data_iter_produced_num_seqs <= self._num_seqs, (
+                        f"{self}: map_seq_stream yielded more seqs ({self._data_iter_produced_num_seqs}) "
+                        f"than expected ({self._num_seqs}). _map_seq_stream_preserves_num_seqs is set to "
```

Comment:
```suggestion
                        f"than expected ({self._num_seqs}). map_seq_stream_preserves_num_seqs is set to "
```

Reply:
`_map_seq_stream_preserves_num_seqs` is only the internal name, which should not be relevant to the user. The user sets (or controls) `map_seq_stream_preserves_num_seqs`.